src/main/java/com/example/bookstoreapp/controller/BookController.java

```diff
@@ -29,19 +31,19 @@ public List<BookDto> getAll() {
     }
 
     @GetMapping("/{id}")
-    public BookDto getById(@PathVariable Long id) {
+    public BookDto getById(@PathVariable @Positive Long id) {
```

Comment:
you need to add `@Validated` annotation on your controller class in order to make validation of `@PathVariable, @RequestParam` etc. possible